File: frontend/src/utils/catalogDefaults.js
const ICON_KEYWORDS = [
  { keywords: ['document', 'documents', 'وثيقة', 'وثائق', 'مستند', 'مستندات'], icon: 'file-text' },
  { keywords: ['payment', 'payments', 'دفع', 'مدفوعات', 'رسوم', 'مالية'], icon: 'credit-card' },
  { keywords: ['order', 'orders', 'طلب', 'طلبات'], icon: 'clipboard-list' },
  { keywords: ['provider', 'providers', 'مزود', 'مزوّد', 'مزودين', 'مزوّدين'], icon: 'briefcase-business' },
  { keywords: ['user', 'users', 'عميل', 'عملاء', 'مستخدم', 'مستخدمين'], icon: 'users-round' },
  { keywords: ['report', 'reports', 'تقرير', 'تقارير'], icon: 'line-chart' },
  { keywords: ['notification', 'notifications', 'اشعار', 'إشعار', 'تنبيه', 'تنبيهات'], icon: 'bell' },
  { keywords: ['support', 'help', 'مساعدة', 'دعم'], icon: 'circle-help' },
]

function buildFallbackHash(source) {
  let hash = 0
  for (const character of source) {
    hash = (hash * 31 + character.charCodeAt(0)) >>> 0
  }
  return hash.toString(16).padStart(8, '0').slice(0, 8)
}

export function generateCatalogSlug(values, defaultPrefix = 'item') {
  const list = Array.isArray(values) ? values : [values]

  for (const value of list) {
    const slug = String(value || '')
      .toLowerCase()
      .trim()
      .replace(/[^a-z0-9]+/g, '-')
      .replace(/^-+|-+$/g, '')

    if (slug) return slug
  }

  const rawSource = list.map((value) => String(value || '').trim()).filter(Boolean).join(' ')
  if (rawSource) {
    return `${defaultPrefix}-${buildFallbackHash(rawSource)}`
  }

  return defaultPrefix
}

export function suggestCategoryIcon(...values) {
  const haystack = values.map((value) => String(value || '').toLowerCase()).join(' ')
  const match = ICON_KEYWORDS.find((entry) => entry.keywords.some((keyword) => haystack.includes(keyword.toLowerCase())))
  return match?.icon || 'folder-tree'
}

File: backend/services/catalog_defaults.py
import hashlib

from django.utils.text import slugify

_ICON_KEYWORDS = (
    (("document", "documents", "وثيقة", "وثائق", "مستند", "مستندات"), "file-text"),
    (("payment", "payments", "دفع", "مدفوعات", "رسوم", "مالية"), "credit-card"),
    (("order", "orders", "طلب", "طلبات"), "clipboard-list"),
    (("provider", "providers", "مزود", "مزوّد", "مزودين", "مزوّدين"), "briefcase-business"),
    (("user", "users", "عميل", "عملاء", "مستخدم", "مستخدمين"), "users-round"),
    (("report", "reports", "تقرير", "تقارير"), "line-chart"),
    (("notification", "notifications", "اشعار", "إشعار", "تنبيه", "تنبيهات"), "bell"),
    (("support", "help", "مساعدة", "دعم"), "circle-help"),
)


def fallback_slug(*parts, default_prefix="item"):
    raw_source = " ".join(str(part or "").strip() for part in parts if str(part or "").strip())
    for part in parts:
        value = slugify(part or "")
        if value:
            return value
    if raw_source:
        digest = hashlib.sha1(raw_source.encode("utf-8")).hexdigest()[:8]
        return f"{default_prefix}-{digest}"
    return default_prefix


def suggest_category_icon(*parts):
    haystack = " ".join(str(part or "").strip().lower() for part in parts)
    for keywords, icon in _ICON_KEYWORDS:
        if any(keyword in haystack for keyword in keywords):
            return icon
    return "folder-tree"
